### PR TITLE
[supervisor] Do not log error when it has Canceled code

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

Do not log error when it has `Canceled` code

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all-ci
- [ ] /werft no-test
      Run Leeway with `--dont-test`
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/supervisor/pkg/serverapi/publicapi.go [hunk 1]
```diff
@@ -339,10 +339,11 @@ func (s *Service) publicAPIInstanceUpdate(ctx context.Context, errChan chan erro
 	for {
 		resp, err := resp.Recv()
 		if err != nil {
-			if err != io.EOF && status.Code(err) != codes.Unavailable {
+			code := status.Code(err)
+			if err != io.EOF && code != codes.Unavailable && code != codes.Canceled {
```
Comment (akosyakov) [inline components/supervisor/pkg/serverapi/publicapi.go:343]: @jeanp413 are you sure it works, I tried the same approach in https://github.com/gitpod-io/gitpod/pull/15795 

but it did not work, it seems that workspacekit wrap cancelled in unknown, i.e. it looks like cancelled in logs, but i practice it is status.Unknown.

I ended up doing `if err != io.EOF && ctx.Err() == nil && status.Code(err) != codes.Unavailable {` and it was enough

**Review by akosyakov — APPROVED** [review]
/hold

code looks good, but i'm not sure it works, i had the same problem in another branch and tried the same thing but it did not work for me

It would be good to have how to test to be sure.

File: components/supervisor/pkg/serverapi/publicapi.go (hunk 1 above)
Comment (jeanp413) [reply components/supervisor/pkg/serverapi/publicapi.go:343]: Will check :+1:

**Comment by mustard-mh** [comment]
Why we filter out cancel error? @jeanp413

**Comment by akosyakov** [comment]
@mustard-mh to clean up log output in supervisor on shutdown

**Pull request merged**